hazelcast/src/main/java/com/hazelcast/spi/CallStatus.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi;
+
+/**
+ *
+ *
+ *
+ * <h1>CallStatus doesn't need ot be enum</h1>
+ * CallStatus is currently an enumeration. But if for whatever reason state needs to be returned, we can easily convert this
+ * enumeration to a regular object like e.g. java https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html.
+ *
+ * So you can have a call status class and if e.g. an operation that requires interleaving, a new Interleaving (subclass of
+ * CallStatus) can be returned containing all the state for that interleaving.
+ *
+ * The same can be done for Blocking operations. Currently we just return WAIT, but this can easily be modified by creating a new
+ * Wait (subclass of CallStatus) containing the wait key or whatever else is needed.
+ *
+ */
+public enum CallStatus {
+
```

Comment:
When I read the enum it bothers me a bit that some statuses indicate that the operations has finished implicitly and some not. It makes it ambiguous, e.g. "NO_RESPONSE", can be read that there is no response yet. 
Could we try to formulate it in a clearer way. Just giving it a shot :

- COMPLETED_HAS_RESPONSE
- COMPLETED_NO_RESPONSE
- OFFLOADED
- SHOULD_WAIT